lib/ops/opsservice/packages.go

```diff
@@ -0,0 +1,156 @@
+}
+
+// teleportNodeConfigPackage generates a new teleport configuration package name for the specified
+// node and planet package version.
+//
+// The package is named as '<cluster-name>/teleport-node-config-<node-addr>:<teleport-version>'
+// where node-addr is a combination of the node address and cluster name
+func (s *site) teleportNodeConfigPackage(node remoteServer) loc.Locator {
+	return teleportNodeConfigPackage(node, s.domainName, s.domainName, s.teleportPackage.Version)
+}
+
+// teleportNextMasterConfigPackage generates a new teleport configuration package name for the specified
+// node and planet package version.
+//
+// The package is named as '<cluster-name>/teleport-master-config-<node-addr>:<teleport-version>(+<increment>)?'
+// where increment is an ever-increasing counter and node-addr is a combination of the node address
+// and cluster name
+func (s *site) teleportNextMasterConfigPackage(master remoteServer, teleportVersion string) loc.Locator {
+	teleportVersion = fmt.Sprintf("%v+%v", teleportVersion, time.Now().UTC().Unix())
+	return teleportMasterConfigPackage(master, s.domainName, s.domainName, teleportVersion)
+}
+
+// teleportMasterConfigPackage generates a new teleport configuration package name for the specified
+// node and planet package version.
+//
+// The package is named as '<cluster-name>/teleport-master-config-<node-addr>:<teleport-version>'
+// where node-addr is a combination of the node address and cluster name
+func (s *site) teleportMasterConfigPackage(master remoteServer) loc.Locator {
+	return teleportMasterConfigPackage(master, s.domainName, s.domainName, s.teleportPackage.Version)
+}
+
+func teleportMasterConfigPackage(master remoteServer, repository, clusterName, teleportVersion string) loc.Locator {
+	return loc.Locator{
+		Repository: repository,
+		Name: fmt.Sprintf("%v-%v",
+			constants.TeleportMasterConfigPackage,
+			PackageSuffix(master, clusterName),
+		),
+		Version: teleportVersion,
+	}
+}
+
+func planetSecretsPackage(node *ProvisionedServer, repository, clusterName, planetVersion string) loc.Locator {
+	return loc.Locator{
+		Repository: repository,
+		Name:       fmt.Sprintf("planet-%v-secrets", node.AdvertiseIP),
+		Version:    planetVersion,
+	}
+}
+
+func planetConfigPackage(node remoteServer, repository, clusterName, planetVersion string) loc.Locator {
+	return loc.Locator{
+		Repository: repository,
+		Name: fmt.Sprintf("%v-%v",
+			constants.PlanetConfigPackage,
+			PackageSuffix(node, clusterName)),
+		Version: planetVersion,
+	}
+}
+
+func teleportNodeConfigPackage(node remoteServer, repository, clusterName, teleportVersion string) loc.Locator {
+	return loc.Locator{
+		Repository: repository,
+		Name: fmt.Sprintf("%v-%v",
+			constants.TeleportNodeConfigPackage,
+			PackageSuffix(node, clusterName),
+		),
+		Version: teleportVersion,
+	}
+}
+
+// suffixer replaces characters unacceptable as a package suffix
+var suffixer = strings.NewReplacer(".", "", ":", "")
```

Comment:
Nit: is it more common to simply remove the character, or replace with an `_` or other substitute character?

Reply:
I don't know which is more common (in what context), but I'd say that replacing with an underscore is less destructive but both approaches will fall short if reversibility is desired.
At least in package names, I see no benefit in replacing with underscore vs removing.